integration-tests/utils/integration-utils.js

```diff
@@ -211,9 +212,45 @@ const dismissNPSSurvey = async page => {
   }
 }
 
+// Test workaround: Retry loading of Terra UI if fails first time. This issue often happens after new deploy to Staging/Alpha.
 const signIntoTerra = async (page, { token, testUrl }) => {
-  !!testUrl && await page.goto(testUrl, navOptionNetworkIdle())
-  await page.waitForXPath('//*[contains(normalize-space(.),"Loading Terra")]', { hidden: true, timeout: 60 * 1000 })
+  console.log('signIntoTerra ...')
+
+  const retryOptions = {
+    factor: 1,
+    minTimeout: 1000, // This is min wait time between retries
+    onFailedAttempt: error => {
+      console.error(
+        `Sign in to Terra attempt ${error.attemptNumber} failed. There are ${error.retriesLeft} retries left.`
+      )
+    },
+    retries: 2
+  }
+
+  const timeout = 60 * 1000
+  const run = async url => {
+    try {
+      const httpResponse = await page.goto(url, { waitUntil: ['networkidle0', 'domcontentloaded'], timeout })
```

Comment:
Why switch away from `navOptionNetworkIdle` here? We use `navOptionNetworkIdle` for other page navigations.
https://github.com/DataBiosphere/terra-ui/blob/8f48397a8bcef3ebffa75b143a249aa5ffd78ae2/integration-tests/utils/integration-utils.js#L396-L398

Reply:
I think it may be a _mistake_ to use `networkidle0` in page navigation/goto. It considers navigation to be finished when there are no more than 0 `fetch` network requests for 500 ms. Since there are other types of network requests triggered automatically when loading `index.html` page, I think it self alone may not be sufficient.

Reply:
I'll change it back. Re `fetch`, I found it [here](https://github.com/puppeteer/puppeteer/issues/1552#issuecomment-350954419). He was one of original Puppeteer developers.